### DISCUSSION TITLE
Feature Request: add argument filename to read_json

### DISCUSSION THREAD

**Discussion by andrewRowlinson** [duckdb/duckdb#6567] [Ideas]
Why do you want this feature?

read_csv has a boolean argument filename, which adds an extra filename column in the results. This is useful when loading multiple CSVs concurrently. It would be useful to add this argument to read_json to populate a table with a column for the filename that it came from.

In my use case, I am loading two separate files that belong to the same soccer match into two different tables:
- https://raw.githubusercontent.com/statsbomb/open-data/master/data/lineups/15946.json
- https://raw.githubusercontent.com/statsbomb/open-data/master/data/events/15946.json

I would like to add a column to the database so that I can join them together.

**Comment by orzom411** [comment]
I'd add to that it would be useful to expose meta data for any file that is opened.  For example the file's creation date or size.  Usually one comes from the prospective of the caller knowing what file(s) are employed in a return set but if we think about a years worth of daily files it could be useful to judge the data based on the file's meta data elements.

**Comment by clementgalopin** [comment]
I am really in need for this feature too :)

**Comment by andrewRowlinson** [comment]
Someone mentioned that this pull request https://github.com/duckdb/duckdb/pull/6912 might resolve this.

**Reply by Mytherin** [reply]
Not yet, that PR makes it easier to add the parameter to the JSON reader but the PR hasn't actually added the parameter yet.

**Comment by andrewRowlinson** [comment]
This feature was added in duckdb version 0.8.0. Should also close #6619 .
